layers_ajax.php

```diff
@@ -65,14 +65,19 @@
 } else if ( $action == 'list' ) {
   // Use JSON to encode our list of layers.
   load_user_layers ( $layer_user, 1 );
-  $ret_layers = array();
+  $ret_layers = [];
   foreach ( $layers as $layer ) {
     user_load_variables ( $layer['cal_layeruser'], 'layer' );
-    $ret_layers[] = array ( 'id' => $layer['cal_layerid'],
+    $ret_layers[] =  ['id' => $layer['cal_layerid'],
       'source' => $layer['cal_layeruser'],
       'color' => $layer['cal_color'],
       'dups' => $layer['cal_dups'],
-      'fullname' => $layerfullname );
+// Craig. This looks like a bug. I can't find $layerfullname set anywhere.
+// And the only table with a fullname is webcal_entry_ext_user, which doesn't seem to be called here.
+// Even $fullname doesn't seem to fit.
+// Is it supposed to be maybe $layer['cal_login'] or $layer['cal_layeruser']?
+
+      'fullname' => $layerfullname];
   }
```

Comment:
The call to user_load_variables with 'layer' as the prefix parameter will cause that method to create a variable named "layer" . "fullname".  So, this code is correct.

You can try the ajax request directly in your browser to see that fullname does get populated in the JSON returned:

http://yourserver/webcalendar/layers_ajax.php?action=list

Here is what I get back:

`{"error":0,"status":"OK","message":"","layers":[{"id":1,"source":"jsmith","color":"#000000","dups":"N","fullname":"Joe Smith"}]}`